lib/rust/ensogl/component/grid-view/src/selectable.rs

```diff
@@ -105,12 +106,28 @@ where
         selection_handler.connect_with_shape(&highlights);
         hover_handler.connect_with_shape(&highlights);
 
-        let grid_frp = grid.as_ref().frp();
+        let grid_ref = grid.as_ref();
+        let grid_frp = grid_ref.frp();
         let network = grid_frp.network();
+        let input = &grid_frp.private.input;
         frp::extend! { network
             eval grid_frp.viewport ([highlights](&vp) {
                 highlight::shape::set_viewport(&highlights, vp);
             });
+
+
+            // === Move selection by one position ===
+
+            input_move_selection <- any(...);
+            input_move_selection <+ input.move_selection_up.constant(Some(Direction::Up));
+            input_move_selection <+ input.move_selection_down.constant(Some(Direction::Down));
+            input_move_selection <+ input.move_selection_left.constant(Some(Direction::Left));
+            input_move_selection <+ input.move_selection_right.constant(Some(Direction::Right));
+            eval input_move_selection ([grid_ref](dir) {
+                if let Some(dir) = *dir {
+                    grid_ref.move_selection_or_emit_movement_out_of_grid_prevented_event(dir);
+                }
+            });
```

Comment:
instead of checking for Some here, you can use `<=` in FRP which iterates over values. If it was Some, it will iterate over its value. If it was None, it will not.,

Reply:
The use of `Option` here is a workaround to avoid implementing `Default` on the `ArrowDirection` enum and still be able to pass `ArrowDirection` as a value in FRP nodes. AFAIU, values in FRP nodes must implement `Default`, and it feels strange to me to implement a `Default` trait for a direction enum - why should one of the directions be default vs. the others? In such case, I believe I can't use `<=` here, as it would try to create an FRP node with an `ArrowDirection` value, and thus be rejected by the compiler. I'd be grateful for a guidance if there's some better way to resolve this (and be able to remove the need for `Option` here), or if choosing one of the directions as an arbitrary default would be a better solution.